packages/next/server/web/adapter.ts

```diff
@@ -23,6 +23,16 @@ export async function adapter(params: {
   const buildId = requestUrl.buildId
   requestUrl.buildId = ''
 
+  const isDataReq =
+    requestUrl.searchParams.has('_nextDataReq') ||
```

Comment:
Should we just rely on headers? is it possible at all?

Reply:
Yeah we should be able to remove this now that we use the header, good catch

Reply:
Updated in https://github.com/vercel/next.js/pull/37574/commits/c65c67f2628f52374163128f182e45222a42aaa8